ec2/spark_ec2.py

```diff
@@ -700,7 +704,8 @@ def get_instances(group_names):
 # Deploy configuration files and run setup scripts on a newly launched
 # or started EC2 cluster.
 def setup_cluster(conn, master_nodes, slave_nodes, opts, deploy_ssh_key):
-    master = master_nodes[0].public_dns_name
+    master = master_nodes[0].public_dns_name if not opts.private_ips else \
```

Comment:
Hello @mdagost! There's a conditional logic being repeated more than 10 times. Probably you might want to extract it in a function. @nchammas